c2corg_api/views/user.py

```diff
@@ -486,15 +490,37 @@ def post(self):
         request = self.request
         username = request.validated['username']
         password = request.validated['password']
+        accept_tos = request.validated['accept_tos']
         user = DBSession.query(User). \
             filter(User.username == username).first()
+
         # try to use the username as email if we didn't find the user
         if user is None and is_valid_email(username):
             user = DBSession.query(User). \
                 filter(User.email == username).first()
 
         token = try_login(user, password, request) if user else None
         if token:
+            # Check if the user has validated Terms of Service, if not,
+            # return a 403 with an explicit message that can be caught
+            # by the frontend
+            if user.tos_validated is None and accept_tos is not True:
+                raise HTTPForbidden('TOS not accepted')
```

Comment:
I haven't checked the other error messages, but do we really need to shorten to `TOS`?
Couldn't `Terms for service not accepted` be used instead?

Reply:
sure